tensorflow/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -2758,6 +2758,10 @@ port::Status MIOpenSupport::DoPrepareForConvolution(
     wrap::miopenSetAllocator(miopen.handle(), nullptr, nullptr, nullptr);
 
     scratch_memory_size = preference.memory;
+
+    // Reuse the scratch_memory_temp in actual convolution
+    *scratch_memory = scratch_memory_temp;
+    return port::Status::OK();
```

Comment:
the logic here is correct,

Can we change the arrangement though, so there is only one `return port::Status::OK()` statement in the routine.

One way would be to drop the return statement at line 2764, and extend the end of the else block at line 2765, from its current ending at line 2769 to line 2788 instead...think that will work...what do you think?

Reply:
I like it, will do.